frontend/src/components/common/preferences-modal/preferences-modal-container.tsx

```diff
@@ -0,0 +1,75 @@
+import { DataStatus } from 'common/enums/enums';
+import { ConfirmationModal } from 'components/common/common';
+import { PreferencesModal } from 'components/common/preferences-modal/preferences-modal';
+import { useAppDispatch, useAppSelector } from 'hooks/hooks';
+import { ReactElement, useCallback, useEffect, useState } from 'react';
+import { UserBindCategoriesDto } from 'shared/build';
+import { bindPreferences, chooseAllPreferences } from 'store/preferences/actions';
+
+const PreferencesModalContainer = (): ReactElement => {
+  const dispatch = useAppDispatch();
+
+  const { choosedPreferences } = useAppSelector((state) => ({
+    choosedPreferences: state.preference,
+  }));
+
+  const [isNeedModal, setIsNeedModal] = useState(false);
+  const [isNeedConfirmModal, setIsNeedConfirmModal] = useState(false);
+
+  useEffect(() => {
+    if (choosedPreferences.dataStatus === DataStatus.FULFILLED) {
+      setIsNeedModal(!choosedPreferences.data.length);
```

Comment:
boolean expected, cast this argument explicitly to boolean (const isEmptyPreferences = Boolean(choosedPreferences.data.length))